modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc

```diff
@@ -0,0 +1,128 @@
+
+If you need to migrate from User Key-based to Master Key-based encryption, there are several steps that you need to follow to ensure a smooth and complete transition:
+
+. xref:disable-user-key-based-encryption[Disable User Key-based encryption]
+. xref:remove-the-encryption-records-from-the-owncloud-database[Remove the encryption records from the ownCloud database]
+. xref:remove-the-files_encryption-directory[Remove the `files_encryption` directory]
+. xref:encrypt-the-filesystem-using-master-key-based-encryption[Encrypt the filesystem using Master Key-based encryption]
+
+
+[[disable-user-key-based-encryption]]
+== Disable User Key-based Encryption
+
+The first part of the migration process is to decrypt all files and to disable encryption in ownCloud, which requires three commands to be executed.
+These commands are: xref:configuration/server/occ_command.adoc#encryption[`occ encryption:decrypt-all` and `occ encryption:disable`], and xref::configuration/server/occ_command.adoc#apps-commands[`occ app:disable`].
+
+You can see an example of calling the commands listed below, configured to require no user interaction.
+
+[source,console]
+----
+sudo -u www-data php occ encryption:decrypt-all --continue=yes && \
+  sudo -u www-data php occ encryption:disable --no-interaction && \
+  sudo -u www-data php occ app:disable --no-interaction encryption
+----
+
+NOTE: The decryption of the files by the central administration requires the current passwords of all users!
+
+[[remove-the-encryption-records-from-the-owncloud-database]]
+== Remove the Encryption Records from the ownCloud Database
+
+Once your ownCloud files are unencrypted, and encryption has been disabled, you need to remove the encryption records from the database.
+There is, currently, no `occ` command to handle this, so it has to be done manually.
+Specifically, you need to remove all records from the `oc_appconfig` table where the `appid` column is set to `encryption`.
+
+In the examples below you can see how to do this using the SQLite database.
+If you are not using SQLite, please use the commands specific to your database vendor.
+
+[NOTE]
+====
+The example code assumes that the path to the SQLite database is `<YOUR/OWNCLOUD/ROOT/DIRECTORY>data/owncloud.database`.
+====
+
+[source,console]
+----
+sqlite3 data/owncloud.database
+sqlite> delete from oc_appconfig where appid='encryption';
+sqlite> select * from oc_appconfig where appid='encryption';
+----
+
+Same for mysql/mariadb:
+
+----
+SELECT * FROM `oc_appconfig` WHERE `appid` LIKE 'encryption'
+----
+
+[[remove-the-files_encryption-directory]]
+== Remove the `files_encryption` Directory
+
+With the database updated, next, the `files_encryption` directory needs to be removed.
+Below is an example of how to do so, to save you time.
+
+[source,console]
+----
+find ./data* -name files_encryption -exec rm -rvf {} \;
+----
+
+[[encrypt-the-filesystem-using-master-key-based-encryption]]
```

Comment:
This explicit section id isn't necessary, as one exactly the same will be auto-generated based on the section header below.

Reply:
You mean I shall remove the ID in brackets [[...]] ?

Reply:
Yes, the entire line. When the docs are generated, Antora will automatically create an id which is exactly the same as this one.